src/test/python/test_pythonparser_3.py

```diff
@@ -0,0 +1,155 @@
+import os
+import pytest
+from xml.etree import ElementTree as ET
+from src.main.python.pythonparser_3 import *
+
+_input_dir = "src/test/python/inputs/code_samples/"
+
+
+@pytest.mark.parametrize(
+    "test_input_dirs",
+    [
+        _input_dir + "small_samples",
+
+        _input_dir + "big_samples",
+
+        _input_dir + "class_samples",
+
+        _input_dir + "corner_cases_samples",
+
+        pytest.param(_input_dir + "fstring_samples",
+                     marks=pytest.mark.skip(reason='fstrings are not supported yet')),
+
+        pytest.param(_input_dir + "python_3_8_samples",
+                     marks=pytest.mark.skip(reason='python 3.8 not supported yet'))
+    ]
+)
+def test_produce_valid_xml(test_input_dirs):
+    """
+    :arg: list of input dirs (pytest parametrize)
+    Checks if parser producing valid xml file
+    :return: None
```

Comment:
Just as an additional comment to @nbirillo review, as soon as you add type annotations as she told you to do, you can remove types from docstrings to not to repeat it twice. For more details see https://stackoverflow.com/questions/44389893/which-is-more-efficient-python-docstrings-or-type-hints.
So it will be like:

```
def test_produce_valid_xml(test_input_dirs: List[str]) -> None:
    """
    Checks if parser producing valid xml file
    """
```

Reply:
Okay got it!